primitives/src/v2/mod.rs

```diff
@@ -123,6 +128,12 @@ impl MallocSizeOf for ValidatorId {
 	}
 }
 
+/// Trait required for type specific indices e.g. `ValidatorIndex` and `GroupIndex`
+pub trait TypeIndex {
+	///Converts `Self` to `usize`.
+	fn into_usize(self) -> usize;
```

Comment:
```suggestion
	/// Returns the index associated to this value.
	fn type_index(&self) -> usize;
```

Or maybe something different, but `into_usize` is really opaque.

Reply:
This trait should be private if possible.

Reply:
Had to be public, had leaking error.

Reply:
It appears in the public interface, so it needs to be public.